nemo/collections/common/tokenizers/aggregate_tokenizer.py

```diff
@@ -0,0 +1,149 @@
+        logging.info(f'Aggregate vocab size: {self.vocab_size}')
+
+        # for compatibility purposes only
+        self.tokenizer = DummyTokenizer(self.vocabulary)
+
+        # lookup tables to speed up token to text
+        self.offset_token_ids_by_token_id, self.tokenizers_by_token_id = self._calculate_offsets()
+
+    def _calculate_offsets(self):
+        offsets = {}
+        tokenizers = {}
+        cur_num = 0
+        tot = len(self.tokenizers_dict)
+        for id in range(len(self.vocabulary)):
+            off_id = id - list(self.token_id_offset.values())[cur_num]
+            if cur_num + 1 < tot:
+                if id >= list(self.token_id_offset.values())[cur_num + 1]:
+                    cur_num += 1
+                    off_id = id - list(self.token_id_offset.values())[cur_num]
+            offsets[id] = off_id
+            tokenizers[id] = list(self.tokenizers_dict.values())[cur_num]
+
+        return offsets, tokenizers
+
+    def text_to_tokens(self, text, lang_id):
+        tokenizer = self.tokenizers_dict[lang_id]
+        return tokenizer.text_to_tokens(text)
+
+    def text_to_ids(self, text, lang_id):
+        tokenizer = self.tokenizers_dict[lang_id]
+        token_ids = tokenizer.text_to_ids(text)
+        token_ids[:] = [t + self.token_id_offset[lang_id] for t in token_ids]
+
+        return token_ids
+
+    def tokens_to_text(self, tokens, lang_id):
+        if isinstance(tokens, np.ndarray):
+            tokens = tokens.tolist()
+
+        tokenizer = self.tokenizers_dict[lang_id]
+        return tokenizer.decode_pieces(tokens)
+
+    def ids_to_text(self, ids):
+        if isinstance(ids, np.ndarray):
+            ids = ids.tolist()
+
+        tokens = []
+        for id in ids:
+            offset_id = self.offset_token_ids_by_token_id[id]
+            tokenizer = self.tokenizers_by_token_id[id]
+            tokens.extend(tokenizer.ids_to_tokens([offset_id]))
+        text = ''.join(tokens).replace('▁', ' ')
+
+        return text
+
+    def token_to_id(self, token, lang_id):
+        tokenizer = self.tokenizers_dict[lang_id]
+        return tokenizer.token_to_id(token) + self.token_id_offset[lang_id]
+
+    def ids_to_tokens(self, ids):
+        tokens = []
+
+        for id in ids:
+            offset_id = self.offset_token_ids_by_token_id[id]
+            tokenizer = self.tokenizers_by_token_id[id]
+            token = tokenizer.ids_to_tokens([offset_id])[0]
+            tokens.append(token)
+
+        return tokens
+
+    def tokens_to_ids(self, tokens: Union[str, List[str]], langs: Union[str, List[str]]) -> Union[int, List[int]]:
+        if isinstance(tokens, str):
+            tokens = [tokens]
+        if isinstance(langs, str):
+            langs = [langs]
+
+        ids = []
+        for i, token in enumerate(tokens):
+            lang_id = langs[i]
+            ids.append(self.token_to_id(token, lang_id))
+        return ids
+
+    @property
+    def vocab(self):
+        return self.vocabulary
```

Comment:
Add a property to get supported language names / IDs (maybe return a dict)

Reply:
done, returning list -- e.g. m.tokenizer.langs returns ['en, 'es']